xarray/core/concat.py

```diff
@@ -12,6 +14,7 @@
 )
 
 import pandas as pd
+from typing_extensions import Literal
```

Comment:
I'm not sure why tests didn't fail here, but unconditionally importing from `typing_extensions` causes `import xarray` to fail on Python >=3.8.

Reply:
#6159 should fix this error. You might get errors for python 3.7 instead though, see #5892.

Reply:
We only run nightly with latest Python, so that's fine. I'll deal with the Py 3.7 removals another day...